project/profiles/capacity4more/modules/c4m/og/c4m_og_invite_users/c4m_og_invite_users.module

```diff
@@ -0,0 +1,100 @@
+}
+
+/**
+ * Return true if the user is allowed to invite and false if he isn't.
+ *
+ * Checks by user role and og_role against group visibility (Public/Private) and
+ * Membership requests setting (Open/Moderated).
+ */
+function c4m_og_invite_users_access($group_type, $gid) {
+  if (_og_invite_access($group_type, $gid)) {
+
+    if (user_access('administer site')) {
+      return TRUE;
+    }
+
+    if (og_user_access($group_type, $gid, 'administrator member')) {
+      return TRUE;
+    }
+
+    // Get request type (open/moderated).
+    $group = entity_load_single($group_type, $gid);
+    $group_request_membership_value = c4m_og_get_group_membership_request($group);
+
+    return $group_request_membership_value == 'open' && og_user_access($group_type, $gid, 'member');
+  }
+
+  return FALSE;
+}
+
+/**
+ * Implements hook_mail_alter().
+ */
+function c4m_og_invite_users_mail_alter(&$message) {
+  if ($message['key'] != 'send_invitation') {
+    return;
+  }
+  module_load_include('inc', 'c4m_og_invite_users', 'c4m_og_invite_users.mail');
+  $key = $message['key'] = 'group_invite_user';
+  $variables = $message['params'];
+
+  $message['subject'] = _c4m_og_invite_users_mail_text($key . '_subject', $variables);
+  $message['body'][0] = _c4m_og_invite_users_mail_text($key . '_body', $variables);
+}
+
+/**
+ * Token callback to add unsafe tokens for user mails.
+ *
+ * This function is used by the token_replace() call at the end of
+ * _user_mail_text() to set up some additional tokens that can be
+ * used in email messages generated by user_mail().
+ *
+ * @param array $replacements
+ *   An associative array variable containing mappings from token names to
+ *   values (for use with strtr()).
+ * @param array $data
+ *   An associative array of token replacement values. If the 'group' element
+ *   exists, it must contain a group node object.
+ * @param mixed $options
+ *   Unused parameter required by the token_replace() function.
+ */
+function c4m_og_invite_users_mail_tokens(array &$replacements, array $data, $options) {
+  global $base_url;
+
+  // Formulate the link for the user to accept/decline the invitation.
+  $og_membership = og_get_membership('node', $data['group']->nid, 'user', $data['invitee']->uid);
+  $hash = og_invite_generate_hash($og_membership);
+  $answer_link = url(og_invite_answer_path_by_hash($data['group']->type, $data['group']->nid, $hash, 'answer'));
+  $answer_link = str_replace("/group/group/", "/group/node/", $answer_link);
+
+  $replacements['[personal:subject]'] = 'Invitation to join the ' . $data['group']->type . ' ' . $data['group']->title;
+  $replacements['[user:member]'] = c4m_user_name($data['inviter']);
+
+  // Make the full name lowercase and replace anything except a-z, 1-9  with
+  // dashes, to create the profile URL.
+  $inviter_profile = strtolower(trim($replacements['[user:member]']));
+  $inviter_profile = preg_replace('@[^a-z0-9-]+@', '-', $inviter_profile);
```

Comment:
@ybaras 
I see that we're using the same `preg_replace('@[^a-z0-9-]+@'` more than once,
it is better taking it out to an external function (also for a better documentation).